app/services/reports/facility_progress_service.rb

```diff
@@ -82,15 +83,88 @@ def dimension_combinations_for(indicator)
 
     attr_reader :diabetes_enabled
 
-    memoize def daily_registrations_grouped_by_day
-      diagnosis = diabetes_enabled ? :all : :hypertension
-      RegisteredPatientsQuery.new.count_daily(facility, diagnosis: diagnosis, last: 30)
+    memoize def daily_total_follow_ups
+      records = Reports::FacilityDailyFollowUpAndRegistration.for_region(region).where("visit_date > ?", 30.days.ago)
+      records.each_with_object({}) do |record, hsh|
+        hsh[record.period] = if region.diabetes_management_enabled?
+          values_at(record, :daily_follow_ups_htn_all) + values_at(record, :daily_follow_ups_dm_all) - values_at(record, :daily_follow_ups_htn_and_dm_all)
+        else
+          values_at(record, :daily_follow_ups_htn_all)
+        end
+      end
+    end
+
+    memoize def daily_total_registrations
+      records = Reports::FacilityDailyFollowUpAndRegistration.for_region(region).where("visit_date > ?", 30.days.ago)
+      records.each_with_object({}) do |record, hsh|
+        hsh[record.period] = if diabetes_enabled
+          values_at(record, :daily_registrations_htn_all) + values_at(record, :daily_registrations_dm_all) - values_at(record, :daily_registrations_htn_and_dm_all)
```

Comment:
Nice! Do we need to add any specs for `daily_total_registrations`?

Reply:
There is a spec for testing this behaviour